### ISSUE TITLE
Libreoffice build broken in master

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#17954]
## Issue description
Libreoffice doesn't build (test failure):

```
$ nix-build -A libreoffice
...
[build BIN] extras
[build MOD] extras
[build ALL] All modules but instset: UnoControls accessibility animations apple_remote avmedia basctl basebmp basegfx basic bean binaryurp bridges canvas chart2 clew cli_ure codemaker comphelper configmgr connectivity cppcanvas cppu cppuhelper cpputools cui dbaccess desktop drawinglayer dtrans editeng embeddedobj embedserv eventattacher extensions external accessories collada2gltf epm hsqldb hyphen libetonyek libfreehand libgltf liblangtag libmspub liborcus libpagemaker libxmlsec lpsolve opencollada rhino ucpp extras filter forms formula fpicker framework helpcompiler hwpfilter i18nlangtag i18npool i18nutil idl idlc io javaunohelper jurt jvmaccess jvmfwk l10ntools librelogo libreofficekit lingucomponent linguistic lotuswordpro o3tl offapi officecfg onlineupdate oovbaapi oox opencl package postprocess pyuno readlicense_oo registry remotebridges reportbuilder reportdesign ridljar rsc sal salhelper sax sc scaddins sccomp scp2 scripting sd sdext setup_native sfx2 shell slideshow smoketest solenv soltools sot starmath stoc store svgio svl svtools svx sw swext sysui test testtools toolkit tools tubes ucb ucbhelper udkapi unodevtools unoidl unoil unotest unotools unoxml ure uui vbahelper vcl winaccessibility wizards writerfilter writerperfect xmerge xmlhelp xmloff xmlreader xmlscript xmlsecurity
[build PRL] CustomTarget/instsetoo_native/install/install.phony
redlineFrame.fodt,fdo69893.odt,fdo70807.odt,fdo75110.odt,fdo75898.odt,shape-textbox.odt,shape-textbox.odt,cp1000071.odt,shape-textbox-vertadjust.odt,shape-textbox-autosize.odt,fdo82191.odt,commented-word.odt,fdo85554.odt,merge-change1.odt,uno-cycle.odt,fdo87448.odt,cp1000115.fodt,tdf90003.odt,tdf97601.odt,tdf83798.odt,tdf90362.fodt,tdf86639.rtf,tdf90883.odt,embedded-data-source.odt,unicodeAltX.odt,draw-anchor-undo.odt,tdf89954.odt,tdf89720.odt,tdf87922.odt,tdf77014.odt,/tmp/nix-build-libreoffice-5.1.5.2.drv-0/libreoffice-5.1.5.2/sw/qa/extras/uiwriter/uiwriter.cxx:2944:SwUiWriterTest::testTdf77014
equality assertion failed
- Expected: 8
- Actual  : 25

tdf92648.docx,tdf96943.odt,tdf88453.odt,tdf88453-table.odt,tdf98987.docx,tdf99004.docx,tdf84695.odt,uiwriter.cxx:2944:Assertion
Test name: SwUiWriterTest::testTdf77014
equality assertion failed
- Expected: 8
- Actual  : 25

Failures !!!
Run: 89   Failure total: 1   Failures: 1   Errors: 0
...
```

http://hydra.nixos.org/build/39050013/nixlog/1/raw

## Technical details

* System: NixOS 16.03
* Nix version: 1.11.2
* Nixpkgs version: 76223aa4b2cc7ebda27a93f00829bd447701740a (latest master as of now)

**Comment by RamKromberg** [comment]
If someone is going to dig through that behemoth of a compile, start with 5.1.4 (https://www.libreoffice.org/about-us/security/advisories/cve-2016-4324/).

**Comment by 7c6f434c** [comment]
>If someone is going to dig through that behemoth of a compile, start with 5.1.4 (https://www.libreoffice.org/about-us/security/advisories/cve-2016-4324/).

Is it even about compile? It is just a test failure, most often the 
problem is lacking something esoteric needed for proper rendering
(fontconfig changes can easily create such problems), some such tests
are just commented using sed, because we lack any idea what the proper
fix for the problem is — you can also comment this test out.

I hope to have time tomorrow to look at this test, probably kill it and
check that the result builds, but feel free to commit commenting out 
yourself (you cannot break it any further…)

**Comment by RamKromberg** [comment]
Let me save you some googling: https://github.com/LibreOffice/core/blob/master/sw/qa/extras/uiwriter/uiwriter.cxx#L3400

And here's the bugzilla if you're curious: https://bugs.documentfoundation.org/show_bug.cgi?id=77014

Personally, I'm pinning an older version anyhow :D

**Comment by 7c6f434c** [comment]
Now that's interesting. The previous 5.1.5.2 build works fine with this test…

**Comment by 7c6f434c** [comment]
@RamKromberg I just killed the test, manually opening the file in the alleged broken version gives correct results anyway…

**Comment by RamKromberg** [comment]
@7c6f434c Oh cool and that all it took? Well, that's good. I guess you'll wait with 5.1.4 for after 16.09?

**Comment by 7c6f434c** [comment]
>@7c6f434c Oh cool and that all it took? Well, that's good. I guess you'll wait with 5.1.4 for after 16.09? 

What do you mean `5.1.4`? I see `5.1.5.2` and `5.2.0.4` in the trunk.
I have fixed both.

I am not completely sure how the release branches are supposed to work
so I do not touch them unless someone says that there is a problem and
the fixes from master don't solve the problem easily.

**Comment by RamKromberg** [comment]
I was curious to know if you'll backport to 16.03 stable (https://github.com/NixOS/nixpkgs-channels/commits/nixos-16.03/pkgs/applications/office/libreoffice) since https://www.libreoffice.org/about-us/security/advisories/cve-2016-4324/ states the issue was resolved in 5.1.4, or will you wait it out since 16.09 coming out.
Just wondering about the releases lifespan really. Nothing specific to libreoffice.

**Comment by 7c6f434c** [comment]
>I was curious to know if you'll backport to 16.03 stable (https://github.com/NixOS/nixpkgs-channels/commits/nixos-16.03/pkgs/applications/office/libreoffice) since https://www.libreoffice.org/about-us/security/advisories/cve-2016-4324/ states the issue was resolved in 5.1.4, or will you wait it out since 16.09 coming out.
>Just wondering about the releases lifespan really. Nothing specific to libreoffice.

No idea about releases lifespan. I guess if you make a PR, it has a good
chance to be merged — maybe even the test will not fail on a release
branch, but I doubt anyone would object to commenting out 
a known-fragile test.

If 16.03 is the latest existing release, it would be strange not to
accept fixes when they are offered…

**Comment by bjornfor** [comment]
@7c6f434c: Thank you! (Since b83dee4a6c7dc66ef47b6ac244d36fe3cdd61796 it works again.)
